packages/expo-updates/android/src/main/java/expo/modules/updates/db/dao/AssetDao.kt

```diff
@@ -34,6 +34,14 @@ abstract class AssetDao {
   )
   abstract fun _unmarkUsedAssetsFromDeletion()
 
+  @Query(
+    "UPDATE assets SET marked_for_deletion = 0 WHERE relative_path IN (" +
```

Comment:
```suggestion
    "DELETE assets WHERE marked_for_deletion = 1 AND WHERE relative_path IN (" +
```
Is it more correct to delete this database row here to avoid it being sent to the pipeline that removes the file as well?

Reply:
resolved in slack, we'll leave this the way it is (keeping the duplicate rows rather than deleting them without returning them in this method)

Reply:
ok, 2 vs 1 is enough for me 😄 i'm already in the process of deploying this change now but will add a follow-up task for next week to switch this logic. (to be clear, the only thing it will affect is when the duplicate rows get deleted, so nothing user facing unless there are a monstrous number of rows in this table.)

Reply:
Yeah. I think this is purely about how tidy we want the table to be rather than any real performance consideration. SQLite could have 10k dummy rows and the effect on disk/RAM/time would be not worth talking about.